supervision/detection/tools/smoother.py

```diff
@@ -0,0 +1,161 @@
+    # remember to `pip install inference`
+    from inference import InferencePipeline
+    import supervision as sv
+
+    box_annotator = sv.BoxAnnotator(color=sv.Color(52, 236, 217))
+    byte_tracker = sv.ByteTrack()
+
+    # Initialize the Smoother
+    smoother = sv.Smoother()
+
+    def render(detections, video_frame):
+        # Parse the detections
+        detections = sv.Detections.from_roboflow(detections)
+
+        # Run a tracker to link predictions across frames
+        detections = byte_tracker.update_with_detections(detections)
+
+        # Record the new frame and get the smoothed predictions
+        smoothed_detections = smoother.update_with_detections(detections)
+
+        # Render
+        image_smoothed = box_annotator.annotate(
+            scene=image.copy(),
+            detections=smoothed_detections
+        )
+
+        # Visualize
+        cv2.imshow("Prediction", image)
+        cv2.waitKey(1)
+
+
+    pipeline = InferencePipeline.init(
+        model_id="microsoft-coco/9", # Or put your custom trained model here
+        # api_key="YOUR_ROBOFLOW_KEY", # Uncomment and fill if you want to access a
+        #                                model that requires auth (or setup a .env file)
+        video_reference=0, # Webcam; can also be video path or RTSP stream
+        on_prediction=render
+    )
+    pipeline.start()
+    pipeline.join()
+    ```
+    """
+
+    def __init__(self, length: int = 5) -> None:
+        """
+        Args:
+            length (int): The current count of detected objects within the zone
+        """
+
+        self.length = length
+
+        self.current_frame = 0
+        self.tracks = {}
+        self.track_ends = {}
+
+    def set_length(self, length: int) -> None:
+        """
+        Sets the number of frames to average out detections over.
+
+        Args:
+            length (int): The number of frames to average out detections over.
+        """
+
+        self.length = length
+        for track_id in self.tracks:
+            self.tracks[track_id] = deque(self.tracks[track_id], maxlen=length)
+
+    def update_with_detections(self, detections: Detections) -> Detections:
+        """
+        Adds a new set of predictions to the smoother. Run this with every new
+        prediction received from the model.
+
+        Args:
+            detections (Detections): The detections to add to the smoother.
+        """
+
+        self.current_frame += 1
+
+        for detection_idx in range(len(detections)):
+            tracker_id = detections.tracker_id[detection_idx]
+            if tracker_id is None:
+                # skip detections without a tracker id
+                continue
+
+            if self.tracks.get(tracker_id, None) is None:
+                self.tracks[tracker_id] = deque(maxlen=self.length)
+
+            self.tracks[tracker_id].append(detections[detection_idx])
+            self.track_ends[tracker_id] = self.current_frame
+
+        for track_id in self.tracks:
```

Comment:
We have no state-clearing mechanism. After some time, most of the tracks we store will be dead.

Reply:
@SkalskiP How would you recommend we clear state?

Reply:
Should we have an explicit method `DetectionsSmoother.clear()`, or some logic here?